### PR TITLE
Add Label for the most three letter ens owner

### PR DESCRIPTION
**Pull request by CreedsCode**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/labels/social/persona/ens/labels_three_letter_ens_count.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{{config(alias='three_letter_ens_count')}}
```
Comment (Hosuke) [inline models/labels/social/persona/ens/labels_three_letter_ens_count.sql:1]: Hi @CreedsCode, please define the model in models/labels/social/labels_social_schema.yml file.
You can follow the pattern in models/labels/social/labels_social_schema.yml

**Comment by Hosuke** [comment]
Maybe this PR is a hackathon submission @andrewhong5297 
[Sample labels](https://dune.com/queries/2140745?d=1)

**Comment by andrewhong5297** [comment]
> Maybe this PR is a hackathon submission @andrewhong5297 
> 
> [Sample labels](https://dune.com/queries/2140745?d=1)

Haha yes it is! They didnt need to PR but very cool they did anyways :) we can revisit this after the hackathon.

File: models/labels/social/persona/ens/labels_three_letter_ens_count.sql (hunk 1 above)
Comment (CreedsCode) [reply models/labels/social/persona/ens/labels_three_letter_ens_count.sql:1]: Added the corresponding model

**Comment by Hosuke** [comment]
Thanks @CreedsCode, I have fix the model name in schema. The spell is lgtm. to run.
Since there is only single address having this label, do we need to merge this PR into spellbook? @andrewhong5297 
https://dune.com/queries/2169365?d=1

**Comment by andrewhong5297** [comment]
maybe we can expand this to be all 3 letter ENS owners instead?

**Comment by CreedsCode** [comment]
@andrewhong5297 Sounds good, should I give it a go?

**Comment by Hosuke** [comment]
> @andrewhong5297 Sounds good, should I give it a go?

Please have a try.

**Comment by CreedsCode** [comment]
> > @andrewhong5297 Sounds good, should I give it a go?
> 
> Please have a try.

lfg I gave it a try in dune first, https://dune.com/queries/2120866.
As I think there is little to change, feel free to prove me wrong!
I will add this to this PR soon.

**Comment by Hosuke** [comment]
> https://dune.com/queries/2120866

LGTM.✅ for the query

File: models/labels/social/persona/ens/labels_three_letter_ens_count.sql (hunk 1 above)
Comment (Hosuke) [reply models/labels/social/persona/ens/labels_three_letter_ens_count.sql:1]: Yes, you have done it correctly.

File: models/labels/social/persona/ens/labels_three_letter_ens_count.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    'most_three_letter_ens_owner' as model_name,
+    'spanish-or-vanish' as contributor,
+    'query' as source,
+    'social' as category,
+    timestamp('2022-03-03') as created_at,
+    now() as updated_at,
+    'personas' as label_type,
+    'The owner of the most three letter ENS Domains'as name
+FROM top
+WHERE owner is not null
```
Comment (Hosuke) [inline models/labels/social/persona/ens/labels_three_letter_ens_count.sql:36]: Maybe we can remove top CTE and rewrite name feild as
`concat('Number of three letter ENS Domains owned', ens_count) as name`
What do you think @CreedsCode 

```suggestion
SELECT
    'blockchain' as blockchain,
    (CONCAT('0x', substring(cast(owner as string), 3))) as address,
    'most_three_letter_ens_owner' as model_name,
    'spanish-or-vanish' as contributor,
    'query' as source,
    'social' as category,
    timestamp('2022-03-03') as created_at,
    now() as updated_at,
    'personas' as label_type,
    concat('Number of three letter ENS Domains owned', ens_count) as name
FROM three_letter_ens_count
WHERE owner is not null
```

**Review by Hosuke — APPROVED** [review]
LGTM.✅ for `labels_three_letter_ens_owner` part.
Thank you @CreedsCode 

[Sample labels_three_letter_ens_owner](https://dune.com/queries/2203475?d=1)

For `labels_three_letter_ens_count` I suggest we can rewrite the labels. See below.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @CreedsCode 

[Sample labels_three_letter_ens_owner](https://dune.com/queries/2203475?d=1)
[Sample labels_three_letter_ens_count](https://dune.com/queries/2217211?d=1)

File: models/labels/social/persona/ens/labels_three_letter_ens_owner_schema.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      sector: labels
+      category: ens
+      contributors: spanish-or-vanish
+    config:
+      tags:
+        ["query", "labels", "ethereum", "ens", "ENS resolver", "ENS reverse"]
+    description: "ENS labels"
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
```
[the remark this answers is not in the document]
Comment (CreedsCode) [reply models/labels/social/persona/ens/labels_three_letter_ens_owner_schema.yml:16]: Yeah sure.

**Review by Hosuke — APPROVED** [review]
Now only `labels_three_letter_ens_count` spell added in this PR.

**Comment by jeff-dude** [comment]
safe to ignore recent CI test failure, as new label was merged but not run in prod yet.

**Pull request merged**